### PR TITLE
Add `Prelude.Text.{lower,upper}ASCII` utilities

### PR DESCRIPTION
**Pull request by Gabriel439**
Related to https://github.com/dhall-lang/dhall-lang/issues/631

In particular, this is based on the suggestion from:

https://github.com/dhall-lang/dhall-lang/issues/631#issuecomment-831476805

### CODE REVIEW

File: Prelude/Text/lowerASCII.dhall [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        , Text/replace "Z" "z"
+        ]
+        Text
+        (λ(replacement : Text → Text) → replacement)
+
+let example0 = assert : lowerASCII "ABCdef" ≡ "abcdef"
+
+let -- This does not lowercase Unicode
+    example1 =
+      assert : "Á" ≡ "Á"
```
Comment (basile-henry) [inline Prelude/Text/lowerASCII.dhall:42]:
```suggestion
      assert : lowerASCII "Á" ≡ "Á"
```

**Comment by ad-si** [comment]
Wouldn't it make sense to also include `upperASCII` while you're already at it?

File: Prelude/Text/lowerASCII.dhall (hunk 1 above)
Comment (TristanCacqueray) [reply Prelude/Text/lowerASCII.dhall:42]: The assertion might already be impressive, but it seems like what @basile-henry suggest is what we need here :-)

**Comment by MonoidMusician** [comment]
You might want to add a warning that decomposed unicode characters (base character + modifier diacritics) may be affected by case conversion:
```dhall
assert : lowerASCII "Á" ≡ "á"
```
(I believe that is correct, but you should test!)

File: Prelude/Text/upperASCII.dhall [hunk 2]
```diff
@@ -1,5 +1,8 @@
 {-|
 Uppercase all ASCII characters
+
+Note that this will also lowercase decomposed Unicode characters that contain
```
Comment (ad-si) [inline Prelude/Text/upperASCII.dhall]: uppercase 😅

File: Prelude/Text/upperASCII.dhall [hunk 3]
```diff
@@ -37,8 +40,17 @@ let upperASCII
 
 let example0 = assert : upperASCII "ABCdef" ≡ "ABCDEF"
 
-let -- This does not uppercase Unicode
+let -- This does not uppercase precomposed Unicode characters
+    --
+    -- • The `Á` in the following example is U+00C1
```
Comment (ad-si) [inline Prelude/Text/upperASCII.dhall:45]: Should be `á`

**Review by ad-si — CHANGES_REQUESTED** [review]

**Pull request merged**